### ISSUE TITLE
Bug: Principal Component Analysis - Eigenvector Matrix is wrong.

### ISSUE THREAD

**Issue by zhuyan166** [accord-net/framework#292]
I computed the same test data using accord-net and R respectively, and got different results.
It seems that Eigenvector Matrix is wrong. The sign of value from even columns is reversed.
Please check the attachment, the spreadsheet which contains the test data and results generated by accord-net and R.

[testDataAndResult.xlsx](https://github.com/accord-net/framework/files/471514/testDataAndResult.xlsx)

**Comment by cesarsouza** [comment]
Hi there,

Thanks for reporting the issue! However, in fact, if it is just the signs of the matrix that are reversed, then the eigenvector matrix is still correct and what you see is not a bug. The Singular Value Decomposition suffers from sign indeterminacy, and the behavior you are experiencing is in fact very common in all statistical packages that provide it. In short, there is no "correct" way to present the signs of the eigenvector matrix and each package will give different results that are up to the their particular implementation.

For a short discussion on why eigenvectors with flipped signs are still valid eigenvectors, please take a look at the [second example in the PrincipalComponentAnalysis documentation page](http://accord-framework.net/docs/html/T_Accord_Statistics_Analysis_PrincipalComponentAnalysis.htm). You can also see the [notes for Scikit-learn's PCA implementation](http://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html), which bring the same notice about eventual sign flipping.

Finally, here is a report on a proposed solution for [solving the SVD sign indeterminacy problem](http://prod.sandia.gov/techlib/access-control.cgi/2007/076422.pdf), but please note that PCA works well with it and I don't know of any packages that concretely implement this technique (for example, MATLAB doesn't - as if it did, [this wouldn't have been required](https://fr.mathworks.com/matlabcentral/fileexchange/22118-sign-correction-in-svd-and-pca)).

Hope it helps!
Cesar

**Comment by zhuyan166** [comment]
Hi Cesar,
Thanks for your quick response.
So,  as you said, Singular Value Decomposition(SVD) suffers sign indeterminacy, but gains higher precision. Right?
I followed the second example in the PrincipalComponentAnalysis documentation page, and created the analysis using the covariance/correlation matrix  by "PrincipalComponentAnalysis.FromCovarianceMatrix/FromCorrelationMatrix". Finally, I got the almost same results with by R, difference between them  <0.0001. That is what I want!
